pynamodb/connection/base.py

```diff
@@ -1442,91 +1143,17 @@ def query(self,
-        # We sort the keys here for determinism. This is mostly done to simplify testing.
-        for key in sorted(expected.keys()):
-            condition = expected[key]
-            if EXISTS in condition:
-                operator = NOT_NULL if condition.get(EXISTS, True) else NULL
-                values = []
-            elif VALUE in condition:
-                operator = EQ
-                values = [condition.get(VALUE)]
-            else:
-                operator = condition.get(COMPARISON_OPERATOR)
-                values = condition.get(ATTR_VALUE_LIST, [])
-            if operator not in QUERY_FILTER_VALUES:
-                raise ValueError("{0} must be one of {1}".format(COMPARISON_OPERATOR, QUERY_FILTER_VALUES))
-            not_contains = operator == NOT_CONTAINS
-            operator = FILTER_EXPRESSION_OPERATOR_MAP[operator]
-            condition = self._get_condition(table_name, key, operator, *values)
-            if not_contains:
-                condition = ~condition
-            if condition_expression is None:
-                condition_expression = condition
-            elif conditional_operator == AND:
-                condition_expression = condition_expression & condition
-            else:
-                condition_expression = condition_expression | condition
-        return condition_expression.serialize(name_placeholders, expression_attribute_values)
-
-    def _get_filter_expression(self, table_name, filters, conditional_operator,
-                               name_placeholders, expression_attribute_values):
-        """
-        Builds the FilterExpression needed for Query and Scan operations
-        """
-        condition_expression = None
-        conditional_operator = conditional_operator[CONDITIONAL_OPERATOR]
-        # We sort the keys here for determinism. This is mostly done to simplify testing.
-        for key in sorted(filters.keys()):
-            condition = filters[key]
-            operator = condition.get(COMPARISON_OPERATOR)
-            if operator not in QUERY_FILTER_VALUES:
-                raise ValueError("{0} must be one of {1}".format(COMPARISON_OPERATOR, QUERY_FILTER_VALUES))
-            not_contains = operator == NOT_CONTAINS
-            operator = FILTER_EXPRESSION_OPERATOR_MAP[operator]
-            values = condition.get(ATTR_VALUE_LIST, [])
-            condition = self._get_condition(table_name, key, operator, *values)
-            if not_contains:
-                condition = ~condition
-            if condition_expression is None:
-                condition_expression = condition
-            elif conditional_operator == AND:
-                condition_expression = condition_expression & condition
-            else:
-                condition_expression = condition_expression | condition
-        return condition_expression.serialize(name_placeholders, expression_attribute_values)
-
     def _get_condition(self, table_name, attribute_name, operator, *values):
```

Comment:
we can probably drop this method and inline the one place we create an `__eq__` expression